### PR TITLE
cli: `wait` flag for use with `deployment status -monitor`

### PR DESCRIPTION
**Pull request by jdockerty**
Addresses comments laid out in https://github.com/hashicorp/nomad/issues/11068

### CODE REVIEW

File: command/deployment_status.go [hunk 1]
```diff
@@ -48,6 +48,9 @@ Status Options:
   -monitor
     Enter monitor mode to poll for updates to the deployment status.
 
+  -wait-time
+    How long to wait before polling an update, used in conjunction with monitor mode. Defaults to 2s.
```
Comment (tgross) [inline command/deployment_status.go:52]: Two requests here: let's change the name to `-wait` as that matches what we use for similar configuration values elsewhere. And let's wrap this to 80col:

```suggestion
-wait
    How long to wait before polling an update, used in conjunction with monitor 
    mode. Defaults to 2s.
```

(And then we'll need to change the actual flag parsing to match.)

**Review by tgross — CHANGES_REQUESTED** [review]
Looks great @jdockerty! 

To get this into mergeable state:
* Address the comments.
* Run `make cl` to add a changelog entry for the flag.
* Add the docs for the flag to the `deployment status` docs in [status.mdx](https://github.com/hashicorp/nomad/blob/main/website/content/docs/commands/deployment/status.mdx)
* Mark it ready for review!

I think at some point someone is going to want to come along and add this feature to `job run` too, but looking at the code for that the monitoring is a bit more complex because we monitor an eval and then switch to monitoring the deployment. What you've got here does the job asked and is worth shipping on its own. 😀

File: command/deployment_status.go (hunk 1 above)
Comment (jdockerty) [reply command/deployment_status.go]: Makes sense :blush: 

I debated myself whether to go with `wait`, but I always seem to land on the more verbose options :smile: This has changed now :rocket:

**Review by tgross — APPROVED** [review]
LGTM! Thanks @jdockerty!

**Pull request merged**